### PR TITLE
Prevents mech RCD from being used with mesons, doubles R-wall deconstruction time

### PR DESCRIPTION
**Pull request by SmArtKar**
## About The Pull Request

Mech RCD has a unique quality of being usable at range, but does not check for LoS which allows you to instantly bore holes in multilayered walls with no warning. R-walls won't save you either, since they can be deconstructed by it just fine. Now, mech RCDs need LoS on turfs you click and deconstructs r-walls twice as slowly (the same time it'd take a normal RCD to deconstruct a normal wall).

## Why It's Good For The Game

mech_rcd

Screen from a recent round - this can be done in 6 seconds, as with correct mech positioning that's as long as you need to deconstruct all the walls and doors. This is incredibly oppressive and I won't be surprised if this started getting abused like hell. This should prevent further issues while still keeping it a good (but not absurdly overpowered) option for when you need to breach into somewhere, since its still quicker (and safer) than thermite.

## Changelog
:cl:
balance: Mech-mounted RCD no longer has wallhacks
balance: Reinforced walls now take double the time to get deconstructed when using an RCD
/:cl:

### CODE REVIEW

**Comment by distributivgesetz** [comment]
make RCDing cancel when the mech moves, that was a problem too as the mech moved into cover after starting

**Comment by SmArtKar** [comment]
> make RCDing cancel when the mech moves, that was a problem too as the mech moved into cover after starting

They're still limited by RCD's range even when going into cover, I just want to get rid of instant drilling

**Comment by Steel-Serperior** [comment]
Can't Durands also just punch through rwalls incredibly fast too?

**Comment by githubuser4141** [comment]
> > make RCDing cancel when the mech moves, that was a problem too as the mech moved into cover after starting
> 
> They're still limited by RCD's range even when going into cover, I just want to get rid of instant drilling

no they're not, in a fast clarke or ripley you can start RCDing something then speed off to the other end of the station and it'll still be RCDing away

**Comment by Loafin34** [comment]
ibuild pr

**Comment by CCC23** [comment]
I got deconstructed pr

**Comment by SmArtKar** [comment]
> > > make RCDing cancel when the mech moves, that was a problem too as the mech moved into cover after starting
> > 
> > 
> > They're still limited by RCD's range even when going into cover, I just want to get rid of instant drilling
> 
> no they're not, in a fast clarke or ripley you can start RCDing something then speed off to the other end of the station and it'll still be RCDing away Or, at least so for Construction

yeah that needs to be fixed then

File: code/game/turfs/closed/wall/reinf_walls.dm [hunk 1]
```diff
@@ -215,9 +215,13 @@
...
 /turf/closed/wall/r_wall/rcd_vals(mob/user, obj/item/construction/rcd/the_rcd)
-	if(the_rcd.canRturf || the_rcd.construction_mode == RCD_WALLFRAME)
+	if (the_rcd.construction_mode == RCD_WALLFRAME)
 		return ..()
-
+	if(!the_rcd.canRturf)
+		return
+	. = ..()
+	if (.)
+		.["delay"] *= RCD_RWALL_DELAY_MULT
```
Comment (SyncIt21) [inline code/game/turfs/closed/wall/reinf_walls.dm:224]: This adds delay to the admin RCD & industrial RCD cause they too can deconstruct rwalls. Is this intended? If so you should ad that to your changelog as well

Comment (SmArtKar) [reply code/game/turfs/closed/wall/reinf_walls.dm:224]: Yeah this is intended, done

**Comment by Ghommie** [comment]
Bro, it seems the other PR closed this branch

**Pull request merged**